tests/phpunit/Configuration/Entry/BadgeTest.php

```diff
@@ -36,16 +36,45 @@
 namespace Infection\Tests\Configuration\Entry;
 
 use Infection\Configuration\Entry\Badge;
+use InvalidArgumentException;
 use PHPUnit\Framework\TestCase;
 
 final class BadgeTest extends TestCase
 {
-    use BadgeAssertions;
-
-    public function test_it_can_be_instantiated(): void
+    public function test_it_can_be_instantiated_with_an_exact_branch_match(): void
     {
         $badge = new Badge('master');
 
-        $this->assertBadgeStateIs($badge, 'master');
+        $this->assertTrue($badge->applicableForBranch('master'));
+        $this->assertFalse($badge->applicableForBranch('mast'));
+        $this->assertFalse($badge->applicableForBranch('master '));
+        $this->assertFalse($badge->applicableForBranch(' master'));
+        $this->assertFalse($badge->applicableForBranch('master1'));
+    }
+
+    public function test_it_can_be_instantiated_with_a_regex_branch_match(): void
+    {
+        $badge = new Badge('/^(foo|bar)$/');
+
+        $this->assertTrue($badge->applicableForBranch('foo'));
+        $this->assertTrue($badge->applicableForBranch('bar'));
+        $this->assertFalse($badge->applicableForBranch('fo'));
+        $this->assertFalse($badge->applicableForBranch('ba'));
+        $this->assertFalse($badge->applicableForBranch('foo '));
+        $this->assertFalse($badge->applicableForBranch(' foo'));
+        $this->assertFalse($badge->applicableForBranch('foo1'));
+    }
+
+    public function test_it_rejects_invalid_regex(): void
+    {
+        try {
+            new Badge('/[/');
+
+            $this->fail();
```

Comment:
the slightly less error prone approach with `expectException()` is probably fine here (and IIRC preferred but don't quote me on this)

Reply:
Ah yes, this is because my initial approach was testing for a previous exception, which I then got rid of.

Indeed, will refactor.

Reply:
Never mind: I still have a previous exception, so this kind of test still applies as-is, and `expectException()` is not a sufficient assertion.